packages/client/test/integration/Resends.test.ts

```diff
@@ -0,0 +1,90 @@
+import 'reflect-metadata'
+import { MessageID, StreamMessage } from 'streamr-client-protocol'
+import { DOCKER_DEV_STORAGE_NODE } from '../../src/ConfigTest'
+import { Stream } from '../../src/Stream'
+import { StreamrClient } from '../../src/StreamrClient'
+import { createClientFactory } from '../test-utils/fake/fakeEnvironment'
+import { createRelativeTestStreamId } from '../test-utils/utils'
+
+describe('Resends', () => {
+
+    describe('waitForStorage', () => {
+
+        let client: StreamrClient
+        let stream: Stream
+
+        beforeEach(async () => {
+            client = createClientFactory().createClient()
+            stream = await client.createStream({
+                id: await createRelativeTestStreamId(module),
+            })
+        })
+
+        it('happy path', async () => {
+            await stream.addToStorageNode(DOCKER_DEV_STORAGE_NODE)
+            const mockId = Date.now()
+            const msg = new StreamMessage({
+                messageId: new MessageID(stream.id, 0, Date.now(), 0, 'publisherId', 'msgChainId'),
+                content: {
+                    mockId
+                }
+            })
+            const publishedMsg = await client.publish(stream.id, msg.getParsedContent())
+            await client.waitForStorage(publishedMsg)
+        })
+
+        it('no match', async () => {
+            await stream.addToStorageNode(DOCKER_DEV_STORAGE_NODE)
+            const mockId = Date.now()
+            const msg = new StreamMessage({
+                messageId: new MessageID(stream.id, 0, Date.now(), 0, 'publisherId', 'msgChainId'),
+                content: {
+                    mockId
+                }
+            })
+            await client.publish(stream.id, msg.getParsedContent())
+            const collectedIds: Set<number> = new Set()
```

Comment:
`Set` is a bit weird here since we don't e.g. publish multiple messages, maybe just a singular value?